core/jazz_environment-event-handler/index.js

```diff
@@ -597,6 +590,239 @@ function triggerBuildJob(payload, serviceDetails, configData) {
+      } else {
+        logger.error("Error creating safe: " + JSON.stringify(response));
+        return reject({
+          "error": "Error creating safe",
+          "details": response.body.message
+        });
+      }
+    });
+
+  });
+}
+
+function deleteSafe(environmentPayload, configData, authToken, envDetails) {
+  return new Promise((resolve, reject) => {
+    if (!configData.TVAULT && !configData.TVAULT.IS_ENABLED) {
+      return resolve({
+        "error": "T-vault is not enabled",
+      });
+    }
+    let safeName;
+    if (envDetails.data && envDetails.data.count > 0 && envDetails.data.environment.length){
+      const envResponse = envDetails.data.environment.filter(ele => ele.logical_id === environmentPayload.logical_id);
+      if (envResponse.length && envResponse[0].metadata && envResponse[0].metadata.safe_details) {
+        safeName = envResponse[0].metadata.safe_details.name;
+      } else {
+        return resolve({
+          "error": environmentPayload.logical_id + ": environment safe details is not found",
+        });
+      }
+    } else {
+      return resolve({
+        "error": environmentPayload.logical_id + ": environment safe details is not found",
+      });
+    }
+    var svcPayload = {
+      uri: `${configData.BASE_API_URL}${configData.TVAULT.API}/${safeName}`,
+      method: "DELETE",
+      headers: {
+        "Authorization": authToken,
+        "Content-Type": "application/json",
+      },
+      rejectUnauthorized: false
+    };
+    request(svcPayload, function (error, response, body) {
+      if (response.statusCode && response.statusCode === 200) {
+        return resolve(body);
+      } else {
+        logger.error("Error deleting safe: " + JSON.stringify(response));
+        return reject({
+          "error": "Error deleting safe",
+          "details": response.body.message
+        });
+      }
+    });
+
+  });
+}
+
+function getAdmins(environmentPayload, configData, authToken, safeName) {
+  return new Promise((resolve, reject) => {
+    var payload = {
+      uri: `${configData.BASE_API_URL}${configData.TVAULT.USERS}`,
+      method: "GET",
+      headers: {
+        "Authorization": authToken,
+        "Content-Type": "application/json"
+      }
+    };
+    request(payload, function (error, response, body) {
+      if (response.statusCode && response.statusCode === 200) {
+        const resultData = {
+          'admins': JSON.parse(body),
+          'safeName': safeName
+        };
+        return resolve(resultData);
+      } else {
+        logger.error("Error getting admins: " + JSON.stringify(response));
+        return reject({
+          "error": "Error getting admins",
+          "details": response.body.message
+        });
+      }
+    });
+
+  });
+}
+
+function addAdminsToSafe(environmentPayload, configData, authToken, res) {
+  return new Promise((resolve, reject) => {
+    var adminsList = res.admins && res.admins.data && res.admins.data.result;
```

Comment:
Should we have a check to see if T-Vault is enabled here?

Reply:
check already provided in wrapping function 'addSafe' of this.